File: IN JAVA/TaxBeans.java
package com.retireup;

public class TaxBeans {

	static float taxbracketsarizona[] = {0, 10000, 25000, 50000, 150000};
	static float taxbracketscalifornia[] = {0, 7582, 17976, 28371, 39384, 49774, 254250, 305100, 1000000};
	static float taxbracketskansas[] = {0, 15000};
	static float taxbracketsfederal[] = {0f, 9276f, 37651f, 91151f, 190151f, 413351f, 415051f};
	static float amountsarizona[] = {0, 259, 691, 1531, 5771};
	static float amountscalifornia[] = {0, 75.82f, 283.70f, 699.50f, 1360.28f, 2191.48f, 21207.75f, 26445.30f, 49429.5f, 109884};
	static float amountskansas[] = {0, 405};
	static float amountsfederal[] = {0, 927.5f, 5183.75f, 18558.75f, 46278.75f, 119934.75f, 120529.75f};
			
	
	public static float[] gettaxBracketArizona(float roi, int year){
		
		if(year == 1 || roi == 0){
			return taxbracketsarizona;
		}
		else{
			if(roi>0){
				for(int i = 1 ; i < taxbracketsarizona.length ; i++){
					taxbracketsarizona[i] = CalculateInflation.calculateInflation(taxbracketsarizona[i], roi);
				}					
			}
			else{
				return taxbracketsarizona;
			}
		}
		
			return taxbracketsarizona;
		
	}
	
	public static float[] getAmountsArizona(int year, float roi, float[] tax_rates){
		
		float temp = 0;
		if(year == 1 || roi == 0){
			
			return amountsarizona;
		}
		
		else if(roi>0){
			
			for(int i=0 ; i<amountsarizona.length-1; i++){
					amountsarizona[i+1] = temp+ CalculateInflation.calculateAmounts(taxbracketsarizona[i], taxbracketsarizona[i+1], tax_rates[i]);
					temp = amountsarizona[i+1];
			}			
		}		
		return amountsarizona;
	}
	
	public static float[] gettax_ratesArizona(){
		
		float tax_rates[] = {2.59f, 2.88f, 3.36f, 4.24f, 4.54f};
		return tax_rates;
	}
	
//  
  //
	//END OF ARIZONA SETTERS AND GETTERS
	  //
		//
	
	public static float[] gettaxBracketCalifornia(float roi, int year){
		
		if(year == 1 || roi == 0){
			return taxbracketscalifornia;
		}
		else{
			if(roi>0){
				for(int i = 1 ; i < taxbracketscalifornia.length ; i++){
					taxbracketscalifornia[i] = CalculateInflation.calculateInflation(taxbracketscalifornia[i], roi);
				}					
			}
			else{
				return taxbracketscalifornia;
			}
		}
		
			return taxbracketscalifornia;
	}
	
	public static float[] getAmountsCalifornia(int year, float roi, float[] tax_rates){
		
		float temp = 0;
		if(year == 1 || roi == 0){
			
			return amountscalifornia;
		}
		
		else if(roi>0){
			System.out.println("hello");
			for(int i=0 ; i<amountscalifornia.length-1; i++){
					amountscalifornia[i+1] = temp+ CalculateInflation.calculateAmounts(taxbracketscalifornia[i], taxbracketscalifornia[i+1], tax_rates[i]);
					temp = amountscalifornia[i+1];
			}			
		}		
		return amountscalifornia;
	}
	
	public static float[] gettax_ratesCalifornia(){
		
		float tax_rates[] = {1, 2, 4, 6, 8, 10.30f, 11.30f, 12.30f, 13.30f};
		return tax_rates;
	}

//
  //
	//END OF CALIFORNIA SETTERS AND GETTERS
	  //
		//
	
	
	public static float[] gettaxBracketKansas(float roi, int year){
		
		//float taxbrackets[] = {0f, 15000f};
		
		if(year == 1 || roi == 0){
			return taxbracketskansas;
		}
		else{
			if(roi>0){
				for(int i = 1 ; i < taxbracketskansas.length ; i++){
					taxbracketskansas[i] = CalculateInflation.calculateInflation(taxbracketskansas[i], roi);
				}					
			}
			else{
				return taxbracketskansas;
			}
		}
		
			return taxbracketskansas;
	}

	
	public static float[] getAmountsKansas(int year, float roi, float[] tax_rates){
		
		float temp = 0;
		if(year == 1 || roi == 0){
			
			return amountskansas;
		}
		
		else if(roi>0){
			
			for(int i=0 ; i<amountskansas.length-1; i++){
					amountskansas[i+1] = temp+ CalculateInflation.calculateAmounts(taxbracketskansas[i], taxbracketskansas[i+1], tax_rates[i]);
					temp = amountskansas[i+1];
			}			
		}		
		return amountskansas;
	}
	
	public static float[] gettax_ratesKansas(){
		
		float tax_rates[] = {2.7f, 4.9f};
		return tax_rates;
	}
	
//
  //
	//END OF KANSAS SETTERS AND GETTERS
	  //
		//
	
	public static float[] gettaxBracketFederal(float roi, int year){
		
		if(year == 1 || roi == 0){
			return taxbracketsfederal;
		}
		else if(roi>0){
				for(int i = 1 ; i < taxbracketsfederal.length ; i++){
					taxbracketsfederal[i] = CalculateInflation.calculateInflation(taxbracketsfederal[i], roi);
				}					
		}
		
			return taxbracketsfederal;
	}
	
	public static float[] getAmountsFederal(int year, float roi, float[] tax_rates){
		
		float temp = 0;
		if(year == 1 || roi == 0){
			
			return amountsfederal;
		}
		
		else if(roi>0){
			
			for(int i=0 ; i<amountsfederal.length-1; i++){
					amountsfederal[i+1] = temp+ CalculateInflation.calculateAmounts(taxbracketsfederal[i], taxbracketsfederal[i+1], tax_rates[i]);
					temp = amountsfederal[i+1];
			}			
		}		
		return amountsfederal;
	}
	
	public static float[] gettax_ratesFederal(){
		
		float tax_rates[] = {10, 15, 25, 28, 33, 35, 39.6f};
		
		return tax_rates;
	}

//
  //
	//END OF FEDERAL SETTERS AND GETTERS
	  //
		//
	
	
}//END OF CLASS

File: Node.js/taxBeans.js
var cal_inflation = require ('./calculateInflation');


tax_brackets_arizona = [0, 10000, 25000, 50000, 150000];
tax_brackets_california = [0, 7582, 17976, 28371, 39384, 49774, 254250, 305100, 1000000];
tax_brackets_kansas = [0, 15000];
tax_brackets_federal = [0, 9276, 37651, 91151, 190151, 413351, 415051];


amounts_arizona = [0, 259, 691, 1531, 5771];
amounts_california = [0, 75.82, 283.70, 699.50, 1360.28, 2191.48, 21207.75, 26445.30, 49429.5, 109884];
amounts_kansas = [0, 405];
amounts_federal = [0, 927.5, 5183.75, 18558.75, 46278.75, 119934.75, 120529.75];


module.exports = {	

//GETTAXBRACKETARIZONA()
	gettaxBracketArizona: function(roi, year){
		
		var length = tax_brackets_arizona.length;
		
			if(year == 1 || roi == 0){
				return tax_brackets_arizona;
			}
			
			//calculating and updating each taxbracket with the inflation rate
			else{
				for(var i=0; i<length; i++){
					tax_brackets_arizona[i] = cal_inflation.calculate_Inflation(tax_brackets_arizona[i], roi);
				}
			}		
		return tax_brackets_arizona;
		
	},

//GETTAXBRACKETCALIFORNIA()
	gettaxBracketCalifornia: function(roi, year){
		
		var length = tax_brackets_california.length;
		
			if(year == 1 || roi == 0){
				return tax_brackets_california;
			}
			
			//calculating and updating each taxbracket with the inflation rate
			else{
				for(var i=1; i<length; i++){
					tax_brackets_california[i] = cal_inflation.calculate_Inflation(tax_brackets_california[i], roi);
				}
			}		
		return tax_brackets_california;
		
	},

//GETTAXBRACKETKANSAS()
	gettaxBracketKansas: function(roi, year){
		
		var length = tax_brackets_kansas.length;
		
			if(year == 1 || roi == 0){
				return tax_brackets_kansas;
			}
			
			//calculating and updating each taxbracket with the inflation rate
			else{
				for(var i=1; i<length; i++){
					tax_brackets_kansas[i] = cal_inflation.calculate_Inflation(tax_brackets_kansas[i], roi);
				}
			}		
		return tax_brackets_kansas;
		
	},
	
//GETTAXBRACKETFEDERAL()
	gettaxBracketFederal: function(roi, year){
		
		var length = tax_brackets_federal.length;
		
			if(year == 1 || roi == 0){
				return tax_brackets_federal;
			}
			else{
				for(var i=0; i<length; i++){
					tax_brackets_federal[i] = cal_inflation.calculate_Inflation(tax_brackets_federal[i], roi);
				}
			}		
		return tax_brackets_federal;
		
	},

//GETTAXAMOUNTSARIZONA()
	gettaxAmountsArizona: function(year, roi, tax_rates){
		
		var temp = 0;
		if(year == 1 || roi == 0){
			return amounts_arizona;
		}
		else{
			
			for(var i=0 ; i<amounts_arizona.length-1; i++){
					amounts_arizona[i+1] = temp + cal_inflation.calculate_Amounts(tax_brackets_arizona[i],tax_brackets_arizona[i+1], tax_rates[i]);
					temp = amounts_arizona[i+1];
			}//end of for
			
		}//end of elseif
		return amounts_arizona;
		
	},

//GETTAXAMOUNTSCALIFORNIA()
	gettaxAmountsCalifornia: function(year, roi, tax_rates){
		
		var temp = 0;
		if(year == 1 || roi == 0){
			return amounts_california;
		}
		
		//calculating and updating amounts, that has to be added with the tax_rate to calculate taxes
		else{
			
			for(var i=0 ; i<amounts_california.length-1; i++){
					amounts_california[i+1] = temp + cal_inflation.calculate_Amounts(tax_brackets_california[i],tax_brackets_california[i+1], tax_rates[i]);
					temp = amounts_california[i+1];
			}//end of for
			
		}//end of elseif
		return amounts_california;
		
	},

//GETTAXAMOUNTSKANSAS()
	gettaxAmountsKansas: function(year, roi, tax_rates){
		var temp = 0;
		if(year == 1 || roi == 0){
			return amounts_kansas;
		}
		
		//calculating and updating amounts, that has to be added with the tax_rate to calculate taxes
		else{
			
			for(var i=0 ; i<amounts_kansas.length-1; i++){
					amounts_kansas[i+1] = temp + cal_inflation.calculate_Amounts(tax_brackets_kansas[i],tax_brackets_kansas[i+1], tax_rates[i]);
					temp = amounts_kansas[i+1];
			}//end of for
			
		}//end of elseif
		return amounts_kansas;
	},
	
//GETTAXAMOUNTSFEDERAL()
	gettaxAmountsFederal: function(year, roi, tax_rates){
		var temp = 0;
		if(year == 1 || roi == 0){
			return amounts_federal;
		}
		
		//calculating and updating amounts, that has to be added with the tax_rate to calculate taxes
		else{
			
			for(var i=0 ; i<amounts_federal.length-1; i++){
					amounts_federal[i+1] = temp + cal_inflation.calculate_Amounts(tax_brackets_federal[i],tax_brackets_federal[i+1], tax_rates[i]);
					temp = amounts_federal[i+1];
			}//end of for
			
		}//end of elseif
		return amounts_federal;
	},

//GETTAXRATESARIZONA()	
	gettax_ratesArizona: function(){
		
		var tax_rates_arizona = [2.59, 2.88, 3.36, 4.24, 4.54];
		
		return tax_rates_arizona;
		
	},

//GETTAXRATESCALIFORNIA()	
	gettax_ratesCalifornia: function(){
		
		var tax_rates_california = [1, 2, 4, 6, 8, 10.30, 11.30, 12.30, 13.30];
		
		return tax_rates_california;
		
	},

//GETTAXRATESKANSAS()	
	gettax_ratesKansas: function(){
		
		var tax_rates_kansas = [2.7, 4.9];
		
		return tax_rates_kansas;
		
	},

//GETTAXRATESFEDERAL()	
	gettax_ratesFederal: function(){
		
		var tax_rates_federal = [10, 15, 25, 28, 33, 35, 39.6];
		
		return tax_rates_federal;
		
	},

};